src/components/Modals/CreateSingleItemModal/ImportStep/ImportStep.tsx

```diff
@@ -44,58 +43,34 @@ export default class ImportStep extends React.PureComponent<Props, State> {
    *
    * @param file - The ZIP file.
    */
-  handleZippedModelFiles = async (file: File) => {
-    const zip: JSZip = await JSZip.loadAsync(file)
-    const fileNames: string[] = []
+  handleZippedModelFiles = async (file: File): Promise<ModelData> => {
+    const loadedFile = await loadFile(file.name, file)
+    const { wearable, content } = loadedFile
 
-    zip.forEach(fileName => {
-      if (!basename(fileName).startsWith('.')) {
-        fileNames.push(fileName)
-      }
-    })
-
-    // asset.json contains data to populate parts of the state
-    const assetJsonPath = fileNames.find(path => basename(path) === ASSET_MANIFEST)
     let assetJson: ItemAssetJson | undefined
-
-    if (assetJsonPath) {
-      const assetRaw = zip.file(assetJsonPath)
-      const content = await assetRaw.async('text')
-      assetJson = JSON.parse(content)
+    let modelPath: string | undefined
```

Comment:
I think we shouldn't keep the concept of `assetJson` and its type, as it refers to the older mechanism of importing assets.

Reply:
Maybe we can expand ModelData (the return type) to return all the properties we currently have on assetJSON which are:

```ts
name
description
rarity
category
bodyShape
```

another option is to just rename the prop?

I'm hesitant to rewrite everything as this also is being checked later on in `getAssetJsonProps`

What did you have in mind?